pkg/defaults/defaults_test.go

```diff
@@ -0,0 +1,211 @@
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+type Fixture struct {
+	t          *testing.T
+	portClient *cli.PortClient
+	stateKey   string
+}
+
+func NewFixture(t *testing.T) *Fixture {
+	stateKey := guuid.NewString()
+	portClient, err := cli.New(config.ApplicationConfig.PortBaseURL, cli.WithHeader("User-Agent", fmt.Sprintf("port-k8s-exporter/0.1 (statekey/%s)", stateKey)),
+		cli.WithClientID(config.ApplicationConfig.PortClientId), cli.WithClientSecret(config.ApplicationConfig.PortClientSecret))
+	if err != nil {
+		t.Errorf("Error building Port client: %s", err.Error())
+	}
+
+	_ = integration.DeleteIntegration(portClient, stateKey)
+	return &Fixture{
+		t:          t,
+		portClient: portClient,
+		stateKey:   stateKey,
+	}
+}
+
+func (f *Fixture) CreateIntegration() {
+	err := integration.NewIntegration(f.portClient, f.stateKey, "", &port.IntegrationAppConfig{
+		Resources: []port.Resource{},
+	})
+
+	if err != nil {
+		f.t.Errorf("Error creating Port integration: %s", err.Error())
+	}
+}
+
+func (f *Fixture) CleanIntegration() {
+	_ = integration.DeleteIntegration(f.portClient, f.stateKey)
+}
+
+func deleteDefaultResources(t *testing.T, portClient *cli.PortClient, stateKey string) {
+	_ = integration.DeleteIntegration(portClient, stateKey)
+	_ = blueprint.DeleteBlueprint(portClient, "workload")
+	_ = blueprint.DeleteBlueprint(portClient, "namespace")
+	_ = blueprint.DeleteBlueprint(portClient, "cluster")
+}
+
+func Test_InitIntegration_InitDefaults(t *testing.T) {
+	f := NewFixture(t)
+	defer deleteDefaultResources(t, f.portClient, f.stateKey)
+	e := InitIntegration(f.portClient, &port.Config{
+		StateKey:               f.stateKey,
+		EventListenerType:      "POLLING",
+		CreateDefaultResources: true,
+	})
+	assert.Nil(t, e)
+
+	_, err := integration.GetIntegration(f.portClient, f.stateKey)
+	assert.Nil(t, err)
+
+	_, err = blueprint.GetBlueprint(f.portClient, "workload")
+	assert.Nil(t, err)
+
+	_, err = blueprint.GetBlueprint(f.portClient, "namespace")
+	assert.Nil(t, err)
+
+	_, err = blueprint.GetBlueprint(f.portClient, "cluster")
+	assert.Nil(t, err)
+}
+
+func Test_InitIntegration_FailingInitDefaults(t *testing.T) {
+	f := NewFixture(t)
+	defer deleteDefaultResources(t, f.portClient, f.stateKey)
+	if _, err := blueprint.NewBlueprint(f.portClient, port.Blueprint{
+		Identifier: "workload",
+		Title:      "Workload",
+		Schema: port.BlueprintSchema{
+			Properties: map[string]port.BlueprintProperty{},
+		},
+	}); err != nil {
+		t.Errorf("Error creating Port blueprint: %s", err.Error())
+	}
+	e := InitIntegration(f.portClient, &port.Config{
+		StateKey:               f.stateKey,
+		EventListenerType:      "POLLING",
+		CreateDefaultResources: true,
+	})
+	assert.Nil(t, e)
+
+	i, err := integration.GetIntegration(f.portClient, f.stateKey)
+	assert.True(t, nil == i.Config.Resources)
+	assert.Nil(t, err)
+
+	_, err = blueprint.GetBlueprint(f.portClient, "namespace")
+	if err != nil {
```

Comment:
You don't need to delete the blueprint, each test starts with clean state

Reply:
its just in case
i dont want the test to fail if one of the tests fail to cleanup